lib/src/data/repository/movie_repository.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:convert';
+import 'package:flutter/services.dart';
+import '../../domain/entity/genre.dart';
+import '../../domain/entity/movie.dart';
+import '../../domain/repository/i_movie_repository.dart';
+import '../model/genre_model.dart';
+
+class MovieRepository implements IMovieRepository {
+  List<Movie>? movies;
+
+  static const String genresJsonPath = 'mock_data/genres.json';
+  static const String moviesJsonPath = 'mock_data/movies.json';
+
+  @override
+  Future<List<Movie>?> getMovies() async {
+    if (movies == null) {
+      await _loadMoviesFromJson();
+    }
+    return movies;
+  }
+
+  Future<List<Movie>?> _loadMoviesFromJson() async {
+    final String genreData = await rootBundle.loadString(genresJsonPath);
+    final String movieData = await rootBundle.loadString(moviesJsonPath);
+
+    final Map<String, dynamic> decodedGenreData =
+        jsonDecode(genreData) as Map<String, dynamic>;
+
+    final List<Map<String, dynamic>> genreList =
+        List.from(decodedGenreData['genres'] as Iterable);
+
+    final List<Genre> genres = GenreModel.fromJson(
+      genreList,
+    )!
+        .toList();
```

Comment:
Why you need to force this nullability on the fromJson

Reply:
hmm, i dont actually. Changed fromJson to use non-nullable types